packages/core/src/anchor/processing-loop.ts

```diff
@@ -107,12 +107,13 @@ export class ProcessingLoop<T> {
           const next = await Promise.race([this.source.next(), rejectOnAbortSignal])
           isDone = next.done
           if (isDone) break
-          const value = next.value
-          if (!value) {
+          const values = next.value
+          if (!values) {
             this.#logger.verbose(`No value received in ProcessingLoop, skipping this iteration`)
             continue
           }
-          await Promise.race([this.handleValue(value), rejectOnAbortSignal])
+          const handlersP = values.map((value) => this.handleValue(value))
+          await Promise.race(handlersP.concat(rejectOnAbortSignal))
```

Comment:
Shouldn't this be 

`await Promise.race(Promise.all(handlersP), rejectOnAbortSignal)`

We want all the handlers to finish, but we want to race between handlers finishing and being aborted.

Reply:
yes!

Reply:
there now.